tests/system/item-namespace-staking.test.ts

```diff
@@ -0,0 +1,702 @@
+          {
+            childUpdatePropagationPermissivenessType: { components: true },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              updatePermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              buildPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              stakingPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+          {
+            childUpdatePropagationPermissivenessType: {
+              freeBuildPermissiveness: true,
+            },
+            inherited: { notInherited: true },
+          },
+        ],
+      },
+      config: {
+        usageRoot: null,
+        usageStateRoot: null,
+        componentRoot: null,
+        usages: null,
+        components: [
+          {
+            mint: craftItemMintKeypair.publicKey,
+            amount: new anchor.BN(1),
+            classIndex: new anchor.BN(itemClassIndex),
+            timeToBuild: null,
+            componentScope: "SCOPE_1",
+            useUsageIndex: 0,
+            condition: { presence: true },
+            inherited: { notInherited: true },
+          },
+        ],
+      },
+    };
+
+    const args = {
+      classIndex: new anchor.BN(itemClassIndex),
+      parentClassIndex: null,
+      space: new anchor.BN(300),
+      desiredNamespaceArraySize: 1,
+      updatePermissivenessToUse: { tokenHolder: true },
+      storeMint: false,
+      storeMetadataFields: false,
+      itemClassData,
+      parentOfParentClassIndex: null,
+    } as Instructions.Item.CreateItemClassArgs;
+
+    const accounts = {
+      itemMint: itemClassMintKeypair.publicKey,
+      parent: null,
+      parentMint: null,
+      parentOfParentClassMint: null,
+      metadataUpdateAuthority: walletKeypair.publicKey,
+      parentUpdateAuthority: null,
+    } as Instructions.Item.CreateItemClassAccounts;
+
+    const additionalArgs =
+      {} as Instructions.Item.CreateItemClassAdditionalArgs;
+
+    await itemProgram.createItemClass(args, accounts, additionalArgs);
+
+    const [itemClassPDA] = await Utils.PDA.getItemPDA(
+      itemClassMintKeypair.publicKey,
+      new anchor.BN(itemClassIndex)
+    );
+
+    const accInfo = await itemProgram.client.account.itemClass.fetch(
+      itemClassPDA,
+      "confirmed"
+    );
+
+    expect(accInfo).toBeDefined();
+  });
+
+  it("Mints craft item NFT", async () => {
+    await mintNFT(provider, craftItemMintKeypair);
+
+    const tokenAccounts =
+      await provider.connection.getParsedTokenAccountsByOwner(
+        walletKeypair.publicKey,
+        {
+          mint: craftItemMintKeypair.publicKey,
+        },
+        "confirmed"
+      );
```

Comment:
I dont think you need to make a network call here, you can just do `getAssociatedTokenAddress` from the @solana/spl-token account, its just a PDA

Reply:
Can we get accountInfo from `getAssociatedTokenAddress`? e.g. token amount and decimals.